### PR TITLE
Add flag to stop secrets polling and backoff exponentially on error

### PR DESCRIPTION
**Pull request by haaawk**

### CODE REVIEW

File: server/src/server.rs [hunk 1]
```diff
@@ -269,16 +269,26 @@ async fn start_builtin_version(server: Arc<Server>) -> Result<()> {
...
     loop {
         if let Err(err) = update_secrets(&server).await {
             if !last_try_was_failure {
                 log::warn!("Could not re-read secrets: {:?}", err);
             }
             last_try_was_failure = true;
+            delay_s *= server.opt.secrets_refresh_exponential_backoff_factor;
+            if delay_s > server.opt.secrets_refresh_max_exponential_backoff_s {
+                delay_s = server.opt.secrets_refresh_max_exponential_backoff_s;
+            }
```
Comment (MarinPostma) [inline server/src/server.rs:282]: one liner:
```suggestion
            delay_s =  delay_s.min(server.opt.secrets_refresh_max_exponential_backoff_s);
```

**Review by MarinPostma — COMMENTED** [review]
lgtm

File: server/src/server.rs (hunk 1 above)
Comment (haaawk) [reply server/src/server.rs:282]: Good idea. Thanks.

File: server/src/server.rs [hunk 2]
```diff
@@ -269,16 +269,26 @@ async fn start_builtin_version(server: Arc<Server>) -> Result<()> {
 
 async fn refresh_secrets(server: Arc<Server>) -> Result<()> {
     let mut last_try_was_failure = false;
+    let mut delay_s = server.opt.secrets_polling_period_s;
```
[the remark this answers is not in the document]
Comment (MarinPostma) [reply server/src/server.rs:272]: or maybe factor it in a `fetch_secrets` function that gets called by both

[the remark this answers is not in the document]
Comment (MarinPostma) [reply server/src/server.rs:272]: We need to start in apply even when it fails to load secrets, otherwise dogfood won't be able to bootstrap I think

**Review by MarinPostma — APPROVED** [review]

File: server/src/server.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (honzasp) [reply server/src/server.rs:272]: Note that for local development, we read the secrets from file, so the only occasion where we read the secrets from the internet is on the managed platform.

**Pull request merged**